### ISSUE TITLE
Remove `chunk` code base from teal

### ISSUE THREAD

**Issue by donyunardi** [insightsengineering/nestdevs-tasks#23]
### Summary
This feature is deprecated in the last release, in favor of `qenv`. 

We have made significant efforts to replace the usage of chunk with qenv in teal modules. However, to ensure a safe transition, we need to release the teal modules with the qenv implementation before proceeding to remove the chunk code base entirely.
* https://github.com/insightsengineering/NEST-roadmap/issues/45
* https://github.com/insightsengineering/NEST-roadmap/issues/69

### Definition of Done
* Remove `chunk` code base from teal framework and modules

### Tasks
* Release teal modules
  We are planning to release tmg, tmosprey, tmgoshawk, tmhermes here: https://github.com/insightsengineering/nestdevs-tasks/issues/25
  But we still need to release tmc.  
* Remove all code base related to `chunk` in:
  * insightsengineering/teal.code/issues/112
  * insightsengineering/teal.transform/issues/129
* insightsengineering/teal.code#102
  * `crayon` is only used in `chunks` feature in `teal.code`. See issue for more details.
* This is a breaking change, make sure the version dependency in DESCRIPTION file is updated for:
  * teal
  * teal.transform
  * the modules (tmc, tmg, tosprey, tgoshawk)
* Test several random module and report if there's any error.
* Double check vignettes for any mention of `chunk`. If found, remove/update.

**Comment by donyunardi** [comment]
@gogonzo 
We can't do this until all teal modules are released, right?

**Comment by gogonzo** [comment]
@donyunardi we've removed `chunks` from all modules and they have been released already without `chunks`. There are just few occurrences of chunks in other packages and they should be removed also (as these elements containing `chunks` are not used also). I wonder what `chunks` does in `teal.modules.hermes` - @danielinteractive ?

Occurrences of the chunks in the codebase:
https://github.com/search?q=org%3Ainsightsengineering%20chunks&type=code

**Comment by danielinteractive** [comment]
@gogonzo Fortunately we don't use `chunks` at all in `teal.modules.hermes` or `teal.modules.helios`. So no concerns from our side.

**Comment by donyunardi** [comment]
@gogonzo 

> we've removed chunks from all modules and they have been released already without chunks

I believe you are referring to the update we implemented to replace the usage of `chunks` with `qenv` in teal modules. 
However, we haven't release another stable version of the modules after the implementation.

Here's an example for tmg where we still have `chunks` occurrences in latest stable release v0.2.15
https://github.com/insightsengineering/teal.modules.general/blob/v0.2.15/R/tm_a_pca.R#L266

If we remove `chunks` before introducing a new stable release of tmg, integration will fail.

Therefore, before we can completely eliminate the `chunks` code base, we will need to release another stable version of tmg, and other teal modules that was affected with `qenv` implementation. Once this new release is available, we can safely proceed with the removal of the `chunks` code.

Please let me know your thought on this.

@danielinteractive
As @gogonzo pointed out, I also see `chunks` occurrences in `teal.modules.hermes` latest main.
Click [here](https://github.com/search?q=org%3Ainsightsengineering+repo%3Ainsightsengineering%2Fteal.modules.hermes++chunks&type=code) to see the two related files.

**Comment by danielinteractive** [comment]
Those are Just design files, which are not part of the R package.

**Comment by donyunardi** [comment]
Thank you, @danielinteractive. I will leave it to you to update that section, but it should not hinder/block the progress of this roadmap. I will close the tmh issue associated with this roadmap.
